js/grid.js

```diff
@@ -215,6 +216,16 @@
     },
 
     /**
+     * Test a tile at a given position for being pinned or history
+     *
+     * @param {Number} aPos Position in sites array
+     */
+    _isHistoricalTile(aPos) {
+      let site = this.sites[aPos];
+      return site && (site.isPinned() || site.link && site.link.type === "history");
+    },
+
+    /**
      * Make sure the correct number of rows and columns are visible
      */
     _resizeGrid() {
```

Comment:
Well, resizeGrid is taking into account the user-set values for grid rows and columns, it's only adding a scrollbar for users who have extra rows that contain history or pinned tiles, etc. I think it's doing more than what a flexbox can do...